javascript/javascript2/week2/hyf-bay/main.js

```diff
@@ -0,0 +1,67 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+console.log(products);
+
+const list = document.querySelector("ul");
+let showedProduts = products;
+
+function renderProducts(productArray) {
+    // your code here
+    for (const product of productArray) {
+        const li = document.createElement("li");
+        const header = document.createElement("h2");
+        header.innerHTML = product.name;
+        const price = document.createElement("span");
+        price.innerHTML = "Price: " + product.price;
+        const rating = document.createElement("span");
+        rating.innerHTML = "Rating: " + product.rating;
+        li.appendChild(header);
+        li.appendChild(price)
+        li.appendChild(document.createElement("br"));
+        li.appendChild(rating);
+        list.appendChild(li);
+    }
+}
+
+renderProducts(products);
+
+const inputName = document.querySelector("#nameFilter");
+const inputPrice = document.querySelector("#priceFilter");
+//filtering by product name
+inputName.addEventListener("keyup", () => {
+    showedProduts = products.filter(listing => listing.name.toLowerCase().includes(inputName.value.toLowerCase()));
+    list.innerHTML = "";
+    renderProducts(showedProduts);
+})
+//filtering by product MAX price
+inputPrice.addEventListener("input", () => {
+    showedProduts = products.filter(listing => listing.price < inputPrice.value);
+    list.innerHTML = "";
+    renderProducts(showedProduts);
+})
+
+//sorting
+const radios = document.getElementsByName('sort');
+radios.forEach(radio => radio.addEventListener('change', changeSort));
+
+function changeSort() {
+    list.innerHTML = "";
+    let sorted;
+    for (const radio of radios) {
+        if (radio.checked) {
+            switch (radio.value) {
+                case "name":
+                    sorted = showedProduts.sort((a, b) => a.name - b.name);
```

Comment:
This code is not working for sorting by Name, try this instead ```
sorted = showedProduts.sort((a, b) => {
						if (a.name.toUpperCase() < b.name.toUpperCase()) {
							return -1;
						}
						if (a.name.toUpperCase() > b.name.toUpperCase()) {
							return 1;
						}
						return 0;
					});
```

Reply:
Thanks changed this in new commit!